### DISCUSSION TITLE
How to build zip.js with webpack 3 (Angular 5)?

### DISCUSSION THREAD

**Discussion by jonatanmartinbabel** [gildas-lormeau/zip.js#298] [Q&A]
ERROR in ./node_modules/@zip.js/zip.js/lib/core/codecs/codec-pool-worker.js
Module parse failed: Unexpected token (110:57)
You may need an appropriate loader to handle this file type.
| 
|       function getWorker(options = {}) {
|               return new Worker(new URL(workerData.scripts[0], import.meta.url), options);
|       }
| 

webpack: Failed to compile.

**Comment by gildas-lormeau** [comment]
Are you using the version 4 of webpack?

**Comment by jonatanmartinbabel** [comment]
I am using version 3.11.0 of webpack (Angular 5)

**Comment by jonatanmartinbabel** [comment]
> I am using version 3.11.0 of webpack (Angular 5)

Should I use another version of zip.js ?
Is there a versions table in the docs?

**Comment by gildas-lormeau** [comment]
You probably need to include a loader to add the support of `import.meta.url` in webpack. Here is the procedure how to do so with webpack 4: https://github.com/gildas-lormeau/zip.js/discussions/229. I hope it will work with webpack 3 as well.

**Comment by jonatanmartinbabel** [comment]
> You probably need to include a loader to add the support of `import.meta.url` in webpack. Here is the procedure how to do so with webpack 4: #229. I hope it will work with webpack 3 as well.

Seems like a good solution for some projects, but it didn't work on mine. 

_btw, I found this info from the webpack-import-meta-loader plugin author_
> 
> https://www.npmjs.com/package/@open-wc/webpack-import-meta-loader
> 
> ![image](https://user-images.githubusercontent.com/91730840/149820205-3314296d-497d-4a0b-b148-08201e3de532.png)
> 
>

**Comment by gildas-lormeau** [comment]
I was not able to include babel in the build script, cf. https://github.com/gildas-lormeau/zip.js/issues/296#issuecomment-1016515411.

**Comment by gildas-lormeau** [comment]
I'm moving this issue in the [Discussions](https://github.com/gildas-lormeau/zip.js/discussions) because it's not a bug in zip.js which uses [rollup](https://rollupjs.org/) to build files while the question is related to webpack 3.

**Comment by gildas-lormeau** [comment]
Note that you should be able a built file from the [dist](https://github.com/gildas-lormeau/zip.js/tree/master/dist) to circumvent this issue.

**Reply by pioneer623** [reply]
Do you - by chance - know how to reference one of the `dist` versions through a `package.json` dependency? I have a build system that I can't modify, but am also seeing the error above that I'd like to work around...

**Reply by gildas-lormeau** [reply]
Update zip.js and try this import:
```js
import * as zip from "@zip.js/zip.js/lib/zip-no-worker.min.js";
```

You won't be able to use web workers though.
